pkg/daemon/daemon.go

```diff
@@ -93,398 +55,304 @@ func New(
-				return err
+				reqLogger.Error(err, "failed to check host state")
+				return ctrl.Result{}, err
 			}
 
-			// if the service doesn't exist we should continue to let the k8s plugin to create the service files
-			// this is only for k8s base environments, for openshift the sriov-operator creates a machine config to will apply
-			// the system service and reboot the node the config-daemon doesn't need to do anything.
-			if !(serviceEnabled && postNetworkServiceEnabled) {
-				sriovResult = &systemd.SriovResult{SyncStatus: consts.SyncStatusFailed,
-					LastSyncError: fmt.Sprintf("some sriov systemd services are not available on node: "+
-						"sriov-config available:%t, sriov-config-post-network available:%t", serviceEnabled, postNetworkServiceEnabled)}
-			} else {
-				sriovResult, err = systemd.ReadSriovResult()
+			if shouldUpdate {
+				reqLogger.Info("updating nodeState with new host status")
+				err = dn.updateSyncState(ctx, desiredNodeState, desiredNodeState.Status.SyncStatus, desiredNodeState.Status.LastSyncError)
 				if err != nil {
-					log.Log.Error(err, "nodeStateSyncHandler(): failed to load sriov result file from host")
-					return err
+					reqLogger.Error(err, "failed to update nodeState new host status")
+					return ctrl.Result{}, err
 				}
 			}
-			if sriovResult.LastSyncError != "" || sriovResult.SyncStatus == consts.SyncStatusFailed {
-				log.Log.Info("nodeStateSyncHandler(): sync failed systemd service error", "last-sync-error", sriovResult.LastSyncError)
 
-				// add the error but don't requeue
-				dn.refreshCh <- Message{
-					syncStatus:    consts.SyncStatusFailed,
-					lastSyncError: sriovResult.LastSyncError,
-				}
-				<-dn.syncCh
-				return nil
-			}
+			return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 		}
+	}
 
-		skipReconciliation, err = dn.shouldSkipReconciliation(dn.desiredNodeState)
+	// set sync state to inProgress, but we don't clear the failed status
+	err = dn.updateSyncState(ctx, desiredNodeState, consts.SyncStatusInProgress, desiredNodeState.Status.LastSyncError)
+	if err != nil {
+		reqLogger.Error(err, "failed to update sync status to inProgress")
+		return ctrl.Result{}, err
+	}
+
+	reqReboot, reqDrain, err := dn.checkOnNodeStateChange(desiredNodeState)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if vars.UsingSystemdMode {
+		// When running using systemd check if the applied configuration is the latest one
+		// or there is a new config we need to apply
+		// When using systemd configuration we write the file
+		systemdConfModified, err := dn.writeSystemdConfigFile(desiredNodeState)
 		if err != nil {
-			return err
+			reqLogger.Error(err, "failed to write systemd config file")
+			return ctrl.Result{}, err
 		}
+		reqDrain = reqDrain || systemdConfModified || !exist
+		// require reboot if drain needed for systemd mode
+		reqReboot = reqReboot || systemdConfModified || reqDrain || !exist
```

Comment:
nit: you can drop : `|| systemdConfModified` and `|| !exist`
they are checked in `reqDrain`